src/Stratis.Features.SQLiteWalletRepository/External/ITransactionsToLists.cs

```diff
@@ -114,12 +114,19 @@ public bool ProcessTransactions(IEnumerable<Transaction> transactions, HashHeigh
                     if (txOut.ScriptPubKey.ToBytes(true)[0] == (byte)OpcodeType.OP_RETURN)
                         continue;
 
-                    foreach (Script pubKeyScript in this.GetDestinations(txOut.ScriptPubKey))
+                    var destinations = this.GetDestinations(txOut.ScriptPubKey);
+
+                    bool isChange = destinations.Any(d => addressesOfInterest.Contains(d, out AddressIdentifier address2) && address2.AddressType == 1);
```

Comment:
`destinations` can be null here and will be if the txOut is a smart contract output.